### PR TITLE
Added support for SpanKind in operations

### LINKED ISSUE #76: Add Operation.SpanKind support
## Requirement - what kind of business use case are you trying to solve?

I ran jaeger grpc-plugin integration tests with this plugin and it failed.

## Problem - what in Jaeger blocks you from solving the requirement?

Integration test failed because this plugin doesn't support jaeger/spanstore.Operation.SpanKind.

### PR DESCRIPTION
**Pull request by EinKrebs**
This closes #76.

### CODE REVIEW

**Comment by EinKrebs** [comment]
@pavolloffay Please take a look at new table structure.

**Comment by EinKrebs** [comment]
This closes #76

File: sqlscripts/local/0001-jaeger-index.sql [hunk 1]
```diff
@@ -4,8 +4,9 @@ CREATE TABLE IF NOT EXISTS %s (
      service LowCardinality(String) CODEC(ZSTD(1)),
      operation LowCardinality(String) CODEC(ZSTD(1)),
      durationUs UInt64 CODEC(ZSTD(1)),
-     tags Array(String) CODEC(ZSTD(1)),
-     INDEX idx_tags tags TYPE bloom_filter(0.01) GRANULARITY 64,
+     tag_keys Array(String) CODEC(ZSTD(1)),
+     tag_values Array(String) CODEC(ZSTD(1)),
```
Comment (Alex-Burmak) [inline sqlscripts/local/0001-jaeger-index.sql:8]: I think it'd be better to use nested structure instead of parallel arrays.  

Something like

```
tags Nested {
  key String CODEC(ZSTD(1)),
  value String CODEC(ZSTD(1)),
}
```

https://clickhouse.tech/docs/ru/sql-reference/data-types/nested-data-structures/nested/ 

And if the number of unique key values is relatively small (< 10 000), consider using `LowCardinality` data type

https://clickhouse.tech/docs/en/sql-reference/data-types/lowcardinality/

File: sqlscripts/replication/0003-jaeger-operations-local.sql [hunk 2]
```diff
@@ -4,9 +4,10 @@ CREATE MATERIALIZED VIEW IF NOT EXISTS %s ON CLUSTER '{cluster}'
             PARTITION BY toYYYYMM(date) ORDER BY (date, service, operation)
             SETTINGS index_granularity=32
         POPULATE
-AS SELECT toDate(timestamp) AS date,
+AS SELECT toDate(timestamp)                                                              AS date,
           service,
           operation,
-          count()           as count
+          count()                                                                        as count,
+          if(has(tag_keys, 'span.kind'), tag_values[indexOf(tag_keys, 'span.kind')], '') as spankind
```
Comment (Alex-Burmak) [inline sqlscripts/replication/0003-jaeger-operations-local.sql:11]: Probably it's worth to create a dedicated column for span kind.

File: config.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-address: tcp://some-clickhouse-server:9000
+address: tcp://sas-es55m23pkwixrbts.db.yandex.net:9440
```
Comment (pavolloffay) [inline config.yaml:1]: Please revert the changes in this file.

File: sqlscripts/replication/0001-jaeger-index-local.sql [hunk 4]
```diff
@@ -5,8 +5,13 @@ CREATE TABLE IF NOT EXISTS %s ON CLUSTER '{cluster}'
     service    LowCardinality(String) CODEC (ZSTD(1)),
     operation  LowCardinality(String) CODEC (ZSTD(1)),
     durationUs UInt64 CODEC (ZSTD(1)),
-    tags       Array(String) CODEC (ZSTD(1)),
-    INDEX idx_tags tags TYPE bloom_filter(0.01) GRANULARITY 64,
+    spankind LowCardinality(String),
```
Comment (pavolloffay) [inline sqlscripts/replication/0001-jaeger-index-local.sql:8]: why is the span.kind treated differently than other tags? Search by any tag key and value should work as well. Could we use that instead?

File: sqlscripts/local/0003-jaeger-operations.sql [hunk 5]
```diff
@@ -8,6 +8,7 @@ AS SELECT
     toDate(timestamp) AS date,
     service,
     operation,
-    count() as count
+    count() as count,
+    if(has(tags.key, 'span.kind'), tags.value[indexOf(tags.key, 'span.kind')], '') as spankind
```
Comment (pavolloffay) [inline sqlscripts/local/0003-jaeger-operations.sql:12]: could you please explain this one? Why are we treating span.kind differently than other tags?

Comment (EinKrebs) [reply sqlscripts/local/0003-jaeger-operations.sql:12]: Span.kind tag is used to define SpanKind field in spanstore.Operation. Spankind column is used to get it easier in GetOperations.

**Review by pavolloffay — APPROVED** [review]
